mlflow/utils/autologging_utils.py

```diff
@@ -533,6 +546,213 @@ def end_session(cls):
+        :param call_kwargs: The keyword arguments passed to the patched ML API call.
+        """
+        _logger.debug(
+            "Invoked patched ML API '%s.%s' for %s autologging with args '%s' and kwargs '%s'",
+            patch_obj,
+            function_name,
+            session.integration,
+            call_args,
+            call_kwargs,
+        )
+
+    def log_patch_function_success(self, session, patch_obj, function_name, call_args, call_kwargs):
+        """
+        Called upon successful termination of a patched ML API associated with an autologging
+        integration (e.g., `sklearn.linear_model.LogisticRegression.fit()`).
+
+        :param session: The `AutologgingSession` associated with the patched ML API call.
+        :param patch_obj: The object (class, module, etc) on which the patched ML API was called.
+        :param function_name: The name of the patched API that was called.
+        :param call_args: The positional arguments passed to the patched ML API call.
+        :param call_kwargs: The keyword arguments passed to the patched ML API call.
+        """
+        _logger.debug(
+            "Patched ML API call '%s.%s' for %s autologging completed successfully. Patched ML"
+            " API was called with args '%s' and kwargs '%s'",
+            patch_obj,
+            function_name,
+            session.integration,
+            call_args,
+            call_kwargs,
+        )
+
+    def log_patch_function_error(
+        self, session, patch_obj, function_name, call_args, call_kwargs, exception
+    ):
+        """
+        Called when execution of a patched ML API associated with an autologging integration
+        (e.g., `sklearn.linear_model.LogisticRegression.fit()`) terminates with an exception.
+
+        :param session: The `AutologgingSession` associated with the patched ML API call.
+        :param patch_obj: The object (class, module, etc) on which the patched ML API was called.
+        :param function_name: The name of the patched API that was called.
+        :param call_args: The positional arguments passed to the patched ML API call.
+        :param call_kwargs: The keyword arguments passed to the patched ML API call.
+        :param exception: The exception that caused the patched ML API call to terminate.
+        """
+        _logger.debug(
+            "Patched ML API call '%s.%s' for %s autologging threw exception. Patched ML API was"
+            " called with args '%s' and kwargs '%s'. Exception: %s",
+            patch_obj,
+            function_name,
+            session.integration,
+            call_args,
+            call_kwargs,
+            exception,
+        )
+
+    def log_original_function_start(
+        self, session, patch_obj, function_name, call_args, call_kwargs
+    ):
+        """
+        Called during the execution of a patched ML API associated with an autologging integration
```

Comment:
Can we simply say `a patched API`?

```suggestion
        Called during the execution of a patched API associated with an autologging integration
```

Maybe it's just me but `ML API` sounds like this:

![Screen Shot 2020-12-21 at 12 55 36](https://user-images.githubusercontent.com/17039389/102737865-e2d42080-438b-11eb-8023-ce51981f9f48.png)

https://nordicapis.com/choosing-the-best-machine-learning-api-for-your-project/

Reply:
Sounds good! Neat link :). I've removed "ML"!